static/js/lib/learning_resources.js

```diff
@@ -27,24 +34,131 @@ export const availabilityLabel = (availability: ?string) => {
   }
 }
 
+export const parseDateFilter = (filter: string) => {
+  const format = /(now)(\+)?(\d+)?([Md])?/
+  const match = format.exec(filter)
+  if (match) {
+    let dt = moment()
+    if (match[3] && match[4]) {
+      dt = dt.add(match[3], match[4] === "d" ? "days" : "months")
+    }
+    return dt
+  }
+}
+
+export const inDateRanges = (run: CourseRun, availabilities: Array<string>) => {
+  let inRange = false
+  availabilities.forEach(availability => {
+    if (AVAILABILITY_MAPPING[availability]) {
+      const from = parseDateFilter(
+        AVAILABILITY_MAPPING[availability].filter.from
+      )
+      const to = parseDateFilter(AVAILABILITY_MAPPING[availability].filter.to)
+      const startDate = run.best_start_date
+        ? moment(run.best_start_date, dateFormat)
+        : null
+      if (
+        ((isNil(startDate) && availability === AVAILABLE_NOW) ||
+          isNil(from) ||
+          // $FlowFixMe: okay to compare moments
+          startDate >= from) &&
+        ((isNil(startDate) && availability === AVAILABLE_NOW) ||
+          isNil(to) ||
+          // $FlowFixMe: okay to compare moments
+          startDate <= to)
+      ) {
+        inRange = true
+      }
+    }
+  })
+  return inRange
+}
+
+export const bestRunLabel = (run: ?CourseRun) => {
+  if (!run) {
+    return AVAILABILITY_MAPPING[AVAILABLE_NOW].label
+  }
+  for (const range in AVAILABILITY_MAPPING) {
+    if (inDateRanges(run, [range])) {
+      return AVAILABILITY_MAPPING[range].label
+    }
```

Comment:
ok, so if I'm reading this right, this will return the first entry `AVAILABILITY_MAPPING` that matches the dates for the course run, not necessarily one that the user has selected. is that the behavior that we want?

like right now if I selected `in the next year` I see a number of other labels too, like `in the next three months`, etc. this makes sense because obviously anything happening within the next three months is also happening in the next three months. but something happening in the next three months could _also_ be labeled, equally accurately although less specifically, `in the next year`.

I thought I remembered a discussion that the selected filter value should be the one we show, but I don't know. what do you think is correct?

Reply:
I was thinking that the earliest time frame label that still fits within the user's selected time frame might be better/more informative, ie "in the next 3 months" is better than "in the next year".  @Ferdi or @pdpinch what do you think?

Reply:
Ferdi said to keep it as is, working as expected.